src/Serialization/main.jl

```diff
@@ -1,9 +1,135 @@
+################################################################################
+# Description of the saving and loading mechanisms
+#
+# We require that any types serialized through OSCAR are registered using the
+# @register_serialization_type macro, like so
+
+# @register_serialization_type NewType "String Representation of type" uses_id uses_params
+
+# The last three arguments are optional and can arise in any order. Passing a String
+# arguement will override how the type is store as a string. The last two are boolean
+# flags. When setting uses_id the object will be stored as a reference and will be
+# referred to throughout the serialization using a UUID. This should typically only
+# be used for types that do not have a fixed normal form for example PolyRing and MPolyRing.
```

Comment:
I don't understand this "fixed normal form" bit...

Reply:
I'll talk about this more in another PR

Reply:
PRs and code are not a good place to discuss this.  There is a paper (by @antonydellavecchia @benlorenz and me) coming up on the arXiv by the end of this week.  That should hopefully answer these questions (or needs to be expanded, if not).